src/registry.rs

```diff
@@ -57,26 +59,37 @@ use std::borrow::Cow;
 /// #                "# EOF\n";
 /// # assert_eq!(expected, String::from_utf8(buffer).unwrap());
 /// ```
-#[derive(Debug)]
 pub struct Registry<M = Box<dyn crate::encoding::text::SendSyncEncodeMetric>> {
```

Comment:
With the `Collector` mechanism I don't think there is need for being abstract over `M` any longer. Using a concrete type (e.g. `enum` over all metrics in `prometheus_client::metrics`) will drastically simplify both the library and its usage.

Reply:
See https://github.com/prometheus/client_rust/pull/100 proposing the above.